### PR TITLE
[Docs] add mesheryctl contributors guide

### LINKED ISSUE #2303: [Docs] mesheryctl contributor guide
**Current State:**
Meshery CLI client has a number of nuances to it like use of cobra, viper, logrus, global flags like verbose and so on.

While [Meshery CLI & Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#) serves as both documentation of existing and design spec for forthcoming commands, new and existing contributors will benefit from a contributor guide.

**Desired State:**
- Create a README.md in the /mesheryctl folder that serves as a contributor guide to this component.
- Add link to this README in the main Meshery CONTRIBUTING.md.

---
**Contributor Resources**
- [Meshery documentation site](https://docs.meshery.io/)
- [Meshery documentation source](https://github.com/layer5io/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)

### PR DESCRIPTION
**Pull request by pottekkat**
**Description**

This PR fixes #2303 | **This PR is in draft**

Added a README.md file in `mesheryctl` for providing a guide to contributors.

- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+`mesheryctl` is the CLI Client for Meshery.
+
+# Contributing
+
+Please refer the [Meshery Contributing Guidelines](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md) for setting up your development environment.
+
+## Building and running `mesheryctl`
+
+The `~/meshery/mesheryctl` folder contains the complete code for `mesheryctl`.
```
Comment (leecalcote) [inline mesheryctl/README.md:11]:
```suggestion
The [`/mesheryctl`](https://github.com/layer5io/meshery/tree/master/mesheryctl) folder contains the complete code for `mesheryctl`.
```

**Comment by navendu-pottekkat** [comment]
@leecalcote Is this going in the right direction? Any changes to make?

File: mesheryctl/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+`mesheryctl` is written in Golang or the Go Programming Language. For development use Go version 1.15+.
+
+After making changes, run `make` in the `mesheryctl` folder to build the binary. You can then use the binary by, say, `./mesheryctl system start`.
+
+## `mesheryctl` command reference
+
+Detailed documentation of the `mesheryctl` commands is available in the [Meshery Docs](https://docs.meshery.io/reference/mesheryctl).
+
+## General guidelines and resources
```
Comment (leecalcote) [inline mesheryctl/README.md:21]: Good to add a link to this design spec somewhere in here.

https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#

File: mesheryctl/README.md (hunk 1 above)
Comment (navendu-pottekkat) [reply mesheryctl/README.md]: @leecalcote Added this change.

File: mesheryctl/README.md (hunk 2 above)
Comment (navendu-pottekkat) [reply mesheryctl/README.md:23]: Added link to CLI design spec

**Comment by navendu-pottekkat** [comment]
@leecalcote Added the guide to the contributing docs.

**Review by leecalcote — APPROVED** [review]
Job well done, @navendu-pottekkat 🥇

**Pull request merged**